core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -449,6 +452,10 @@ class GroupMetadataManager(brokerId: Int,
                        | Errors.INVALID_FETCH_SIZE =>
                     Errors.INVALID_COMMIT_OFFSET_SIZE
 
+                  case Errors.INVALID_PRODUCER_ID_MAPPING
+                       | Errors.INVALID_TXN_STATE =>
+                    Errors.UNKNOWN_MEMBER_ID
```

Comment:
Could you please explain the rational behind using UNKNOWN_MEMBER_ID here? We should also add a comment about it because this is not obvious.

Reply:
Sorry I resolved the PR comment, but I had a discussion with myself here: https://github.com/apache/kafka/pull/14370#discussion_r1322225993

Similar to the produce request, we wanted to have a non-fatal error here. (I suppose there is an argument for the invalid PID mapping being fatal) 
What do you think? There are only 2 abortable errors returned by this request. 

```
                } else if (error == Errors.UNKNOWN_MEMBER_ID
                        || error == Errors.ILLEGAL_GENERATION) {
                    abortableError(new CommitFailedException("Transaction offset Commit failed " +
                        "due to consumer group metadata mismatch: " + error.exception().getMessage()));
 ```

Reply:
For posterity, we decided to not go with the abortable error and stick with the fatal one since it makes it clearer what happened. We also can't guarantee clients treat unknown member as non-fatal.